hub/api/dataset.py

```diff
@@ -458,8 +459,15 @@ def copy(
         """
         if isinstance(src, str):
             src_ds = hub.load(src, read_only=True, creds=src_creds, token=src_token)
+        elif isinstance(src, pathlib.Path):
+            src = str(src)
+            src_ds = hub.load(src, read_only=True, creds=src_creds, token=src_token)
```

Comment:
we can just convert the src and prevent having 2 separate calls to hub.load

Reply:
remove if else, use function directly

Reply:
I don't get how can I remove it, could you please explain? src can be of the 3 type (src, pathlib.Path, Dataset). If the type of src is Dataset then it is not possible to use convert_pathlib_to_string_if_needed(). In such scenario I have to convert Dataset to string manually.

Reply:
Ah I see, makes sense then